diff --git a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
index b7908e9..a01430e 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
@@ -152,6 +152,7 @@ let CredentialsApi = class CredentialsApi {
             autoAcceptCredential: options.autoAcceptCredential,
             comment: options.comment,
             connectionRecord,
+            parentThreadId: options.parentThreadId
         });
         this.logger.debug('Offer Message successfully created; message= ', message);
         const outboundMessageContext = await (0, getOutboundMessageContext_1.getOutboundMessageContext)(this.agentContext, {
@@ -248,6 +249,7 @@ let CredentialsApi = class CredentialsApi {
             credentialFormats: options.credentialFormats,
             comment: options.comment,
             autoAcceptCredential: options.autoAcceptCredential,
+            parentThreadId: options.parentThreadId
         });
         this.logger.debug('Offer Message successfully created', { message });
         return { message, credentialRecord };
diff --git a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApiOptions.d.ts b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApiOptions.d.ts
index 5f62b59..c819f9e 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApiOptions.d.ts
+++ b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApiOptions.d.ts
@@ -45,6 +45,7 @@ export interface NegotiateCredentialProposalOptions<CPs extends CredentialProtoc
 export interface CreateCredentialOfferOptions<CPs extends CredentialProtocol[] = CredentialProtocol[]> extends BaseOptions {
     protocolVersion: CredentialProtocolVersionType<CPs>;
     credentialFormats: CredentialFormatPayload<CredentialFormatsFromProtocols<CPs>, 'createOffer'>;
+    parentThreadId?: string;
 }
 /**
  * Interface for CredentialsApi.offerCredential. Extends CreateCredentialOfferOptions, will send an offer
diff --git a/node_modules/@credo-ts/core/build/modules/credentials/protocol/BaseCredentialProtocol.js b/node_modules/@credo-ts/core/build/modules/credentials/protocol/BaseCredentialProtocol.js
index 5b1b54c..67791b3 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/protocol/BaseCredentialProtocol.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/protocol/BaseCredentialProtocol.js
@@ -23,6 +23,11 @@ class BaseCredentialProtocol {
         const credentialRecord = await this.getByProperties(agentContext, {
             threadId: credentialProblemReportMessage.threadId,
         });
+
+        const parentThreadId = credentialProblemReportMessage.thread?.parentThreadId;
+        if (parentThreadId) {
+            credentialRecord.metadata.set('parentThreadId', parentThreadId);
+        }
         // Update record
         credentialRecord.errorMessage = `${credentialProblemReportMessage.description.code}: ${credentialProblemReportMessage.description.en}`;
         await this.updateState(agentContext, credentialRecord, CredentialState_1.CredentialState.Abandoned);
diff --git a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
index b519694..755b442 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
@@ -226,7 +226,7 @@ class V2CredentialProtocol extends BaseCredentialProtocol_1.BaseCredentialProtoc
      * @returns Object containing offer message and associated credential record
      *
      */
-    async createOffer(agentContext, { credentialFormats, autoAcceptCredential, comment, goal, goalCode, connectionRecord, }) {
+    async createOffer(agentContext, { credentialFormats, autoAcceptCredential, comment, goal, goalCode, connectionRecord, parentThreadId, }) {
         const credentialRepository = agentContext.dependencyManager.resolve(repository_1.CredentialRepository);
         const formatServices = this.getFormatServices(credentialFormats);
         if (formatServices.length === 0) {
@@ -248,6 +248,10 @@ class V2CredentialProtocol extends BaseCredentialProtocol_1.BaseCredentialProtoc
             goal,
             goalCode,
         });
+        if (parentThreadId) {
+            offerMessage.setThread({ threadId: credentialRecord.threadId, parentThreadId });
+            credentialRecord.metadata.set('parentThreadId', parentThreadId);
+        }
         agentContext.config.logger.debug(`Saving record and emitting state changed for credential exchange record ${credentialRecord.id}`);
         await credentialRepository.save(agentContext, credentialRecord);
         this.emitStateChangedEvent(agentContext, credentialRecord, null);
@@ -628,6 +632,13 @@ class V2CredentialProtocol extends BaseCredentialProtocol_1.BaseCredentialProtoc
             role: models_1.CredentialRole.Issuer,
             connectionId: connection === null || connection === void 0 ? void 0 : connection.id,
         });
+
+        const parentThreadId = ackMessage.thread?.parentThreadId;
+
+        if (parentThreadId) {
+            credentialRecord.metadata.set('parentThreadId', parentThreadId);
+        }
+
         credentialRecord.connectionId = connection === null || connection === void 0 ? void 0 : connection.id;
         const requestMessage = await didCommMessageRepository.getAgentMessage(messageContext.agentContext, {
             associatedRecordId: credentialRecord.id,
