diff --git a/node_modules/@credo-ts/tenants/build/context/TenantAgentContextProvider.js b/node_modules/@credo-ts/tenants/build/context/TenantAgentContextProvider.js
index a7ed66b..b3acfe1 100644
--- a/node_modules/@credo-ts/tenants/build/context/TenantAgentContextProvider.js
+++ b/node_modules/@credo-ts/tenants/build/context/TenantAgentContextProvider.js
@@ -38,13 +38,13 @@ let TenantAgentContextProvider = class TenantAgentContextProvider {
         }
         // TODO: maybe we can look at not having to retrieve the tenant record if there's already a context available.
         this.logger.debug('debug ========= Get tenantRecord from cache')
-        let tenantRecord = await this.catche.config.cache.get(this.rootAgentContext, `contextCorrelationId-${contextCorrelationId}`)
+        let tenantRecord = await this.cache.config.cache.get(this.rootAgentContext, `contextCorrelationId-${contextCorrelationId}`)
         if(!tenantRecord) {
             
             this.logger.debug('debug ========= TenantRecord not found in cache')
             // TODO: maybe we can look at not having to retrieve the tenant record if there's already a context available.
             tenantRecord = await this.tenantRecordService.getTenantById(this.rootAgentContext, contextCorrelationId)
-            await this.catche.config.cache.set(this.rootAgentContext,`contextCorrelationId-${contextCorrelationId}`,tenantRecord)
+            await this.cache.config.cache.set(this.rootAgentContext,`contextCorrelationId-${contextCorrelationId}`,tenantRecord)
             this.logger.debug(`debug ========= Cached tenant agent context for tenant '$
                 {contextCorrelationId}'`)
           }
