diff --git a/node_modules/@credo-ts/core/build/modules/credentials/formats/jsonld/JsonLdCredentialFormat.d.ts b/node_modules/@credo-ts/core/build/modules/credentials/formats/jsonld/JsonLdCredentialFormat.d.ts
index d12468b..ae70f36 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/formats/jsonld/JsonLdCredentialFormat.d.ts
+++ b/node_modules/@credo-ts/core/build/modules/credentials/formats/jsonld/JsonLdCredentialFormat.d.ts
@@ -10,6 +10,8 @@ export interface JsonCredential {
     issuanceDate: string;
     expirationDate?: string;
     credentialSubject: SingleOrArray<JsonObject>;
+    //TODO change type
+    prettyVc?: any;
     [key: string]: unknown;
 }
 /**
diff --git a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
index fb1fb9d..b519694 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
@@ -97,7 +97,6 @@ class V2CredentialProtocol extends BaseCredentialProtocol_1.BaseCredentialProtoc
         let credentialRecord = await this.findByProperties(messageContext.agentContext, {
             threadId: proposalMessage.threadId,
             role: models_1.CredentialRole.Issuer,
-            connectionId: connection === null || connection === void 0 ? void 0 : connection.id,
         });
         const formatServices = this.getFormatServicesFromMessage(proposalMessage.formats);
         if (formatServices.length === 0) {
diff --git a/node_modules/@credo-ts/core/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js b/node_modules/@credo-ts/core/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
index 9fee65c..80acfe9 100644
--- a/node_modules/@credo-ts/core/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
+++ b/node_modules/@credo-ts/core/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
@@ -74,7 +74,6 @@ let DifPresentationExchangeService = class DifPresentationExchangeService {
         });
         if (errors) {
             const errorMessages = this.formatValidated(errors);
-            console.log("inside validatePresentation DifPresentationExchangeService.js errors ::" ,errorMessages )
             if (errorMessages.length > 0) {
                 throw new DifPresentationExchangeError_1.DifPresentationExchangeError(`Invalid presentation`, { additionalMessages: errorMessages });
             }
diff --git a/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js b/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
index 501bd61..da56801 100644
--- a/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
+++ b/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
@@ -170,8 +170,8 @@ class DifPresentationExchangeProofFormatService {
         try {
             ps.validatePresentationDefinition(request.presentation_definition);
             ps.validatePresentationSubmission(jsonPresentation.presentation_submission);
-            // FIXME: Commenting validatePresentation() for now due to intermittent abandoned issue.
-            // ps.validatePresentation(request.presentation_definition, parsedPresentation);
+             // FIXME: Commenting validatePresentation() for now due to intermittent abandoned issue
+            //ps.validatePresentation(request.presentation_definition, parsedPresentation);
             let verificationResult;
             // FIXME: for some reason it won't accept the input if it doesn't know
             // whether it's a JWT or JSON-LD VP even though the input is the same.
diff --git a/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/ProofFormatCoordinator.js b/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/ProofFormatCoordinator.js
index c749043..83277c5 100644
--- a/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/ProofFormatCoordinator.js
+++ b/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/ProofFormatCoordinator.js
@@ -263,7 +263,6 @@ class ProofFormatCoordinator {
                 formatVerificationResults.push(isValid);
             }
             catch (error) {
-                console.log("inside  processPresentation catch block of  ProofFormatCoordinator.js :: error" ,error)
                 return {
                     message: error.message,
                     isValid: false,
diff --git a/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/V2ProofProtocol.js b/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/V2ProofProtocol.js
index e147559..382c097 100644
--- a/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/V2ProofProtocol.js
+++ b/node_modules/@credo-ts/core/build/modules/proofs/protocol/v2/V2ProofProtocol.js
@@ -513,7 +513,6 @@ class V2ProofProtocol extends BaseProofProtocol_1.BaseProofProtocol {
             message: presentationMessage,
         });
         proofRecord.isVerified = result.isValid;
-        console.log("inside V2ProofProtocol.js result::" , result)
         if (result.isValid) {
             await this.updateState(messageContext.agentContext, proofRecord, models_2.ProofState.PresentationReceived);
         }
